src/app/services/store.service.ts

```diff
@@ -18,7 +20,16 @@ export class StoreService {
     return this.store.select(state => state.categories.loading);
   }
 
-  loadCategories(): void {    
-    this.store.dispatch(CategoryActions.loadCategories());
+  loadCategories(): Observable<CategoryModel[]> {    
+    return this.getCategories().pipe(
+      takeUntilDestroyed(this.destroyRef),
+      switchMap((categories) => {
```

Comment:
Hmm, I think it would be easier to change the effect to make this done. You can leave the dispatch as it was, but in the effects use this.store.select() to get products if they are present